### PR TITLE
Add code coverage check and codecov.io badge

### LINKED ISSUE #251: Code coverage
#### Issue Summary

Add code coverage using [coverage.py](https://coverage.readthedocs.io/en/coverage-4.2/) and [https://codecov.io](https://codecov.io)

#### Steps to Reproduce

1. Install coverage 
`pip install coverage`
2.  Create `.coveragerc`
```cat .coveragerc 
[run]
branch = True
source = sendgrid, tests
omit = site-packages
```
2. Run tests with coverage `coverage run -m unittest`
3. See `coverage report`
```
Name                                   Stmts   Miss Branch BrPart  Cover
------------------------------------------------------------------------
sendgrid/__init__.py                       3      0      0      0   100%
sendgrid/helpers/__init__.py               0      0      0      0   100%
sendgrid/helpers/inbound/__init__.py       2      0      0      0   100%
sendgrid/helpers/inbound/app.py           23      5      4      1    70%
sendgrid/helpers/inbound/config.py        31      4      8      2    74%
sendgrid/helpers/inbound/parse.py         69     42     22      1    33%
sendgrid/helpers/inbound/send.py          28     28      0      0     0%
sendgrid/helpers/mail/__init__.py          1      0      0      0   100%
sendgrid/helpers/mail/mail.py            532     39    196     53    87%
sendgrid/sendgrid.py                      22      2      0      0    91%
sendgrid/version.py                        2      0      2      0   100%
------------------------------------------------------------------------
TOTAL                                    713    120    232     57    79%
```
4. Also you can add https://codecov.io to run on each PR

#### Technical details:

* sendgrid-python Version: master
* Python Version: 3.5

### PR DESCRIPTION
**Pull request by meahow**
This should close #251 
You might want to create separate codecov.io account for sendgrid-python project.
After this, the badge url should be changed in README.md

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=352) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=352) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-python?pullRequest=352) it.</sub>

**Comment by meahow** [comment]
The travis job is broken for py2.6

**Comment by thinkingserious** [comment]
Hi @meahow,

We are dropping support for Python 2.6 in [the next version of this SDK](https://github.com/sendgrid/sendgrid-python/issues/347).

I will still be able to merge your work.

Thanks!

With Best Regards,

Elmer

**Comment by mbernier** [comment]
I attempted to merge the conflicts that popped up, hopefully it builds properly now!

File: README.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [![Travis Badge](https://travis-ci.org/sendgrid/sendgrid-python.svg?branch=master)](https://travis-ci.org/sendgrid/sendgrid-python)
+[![codecov](https://codecov.io/gh/meahow/sendgrid-python/branch/issue_251_coverage/graph/badge.svg)](https://codecov.io/gh/meahow/sendgrid-python)
```
[the remark this answers is not in the document]
Comment (mbernier) [reply README.md]: @thinkingserious can you please create this with the SendGridDX account?

**Comment by thinkingserious** [comment]
Hello @meahow,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/announcing-hacktoberfest-2017">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/swag">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by meahow** [comment]
Thanks @thinkingserious, I just did.

**Pull request merged**